Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,780 @@
+    private var pendingRequests = [String: URLRequest]()
+    
+    private let tab = Tab(configuration: WKWebViewConfiguration().then {
+        $0.processPool = WKProcessPool()
+        $0.preferences = WKPreferences()
+        $0.preferences.javaScriptCanOpenWindowsAutomatically = false
+        $0.allowsInlineMediaPlayback = true
+        $0.ignoresViewportScaleLimits = true
+        //$0.mediaTypesRequiringUserActionForPlayback = []
+    }, type: .private).then {
+        $0.createWebview()
+        $0.webView?.scrollView.layer.masksToBounds = true
+    }
+    
+    private let PlaylistDetectorScript: WKUserScript? = {
+        guard let path = Bundle.main.path(forResource: "PlaylistDetector", ofType: "js"), let source = try? String(contentsOfFile: path) else {
+            log.error("Failed to load PlaylistDetector.js")
+            return nil
+        }
+        
+        var alteredSource = source
+        let token = UserScriptManager.securityToken.uuidString.replacingOccurrences(of: "-", with: "", options: .literal)
+        
+        let replacements = [
+            "$<PlaylistDetector>": "PlaylistDetector_\(token)",
+            "$<security_token>": "\(token)",
+            "$<sendMessage>": "playlistDetector_sendMessage_\(token)",
+            "$<handler>": "playlistCacheLoader_\(UserScriptManager.messageHandlerTokenString)",
+            "$<notify>": "notify_\(token)",
+            "$<onLongPressActivated>": "onLongPressActivated_\(token)",
+            "$<setupLongPress>": "setupLongPress_\(token)",
+            "$<setupDetector>": "setupDetector_\(token)",
+            "$<notifyNodeSource>": "notifyNodeSource_\(token)",
+            "$<notifyNode>": "notifyNode_\(token)",
+            "$<observeNode>": "observeNode_\(token)",
+            "$<observeDocument>": "observeDocument_\(token)",
+            "$<observeDynamicElements>": "observeDynamicElements_\(token)",
+            "$<getAllVideoElements>": "getAllVideoElements_\(token)",
+            "$<getAllAudioElements>": "getAllAudioElements_\(token)",
+            "$<onReady>": "onReady_\(token)",
+            "$<observePage>": "observePage_\(token)",
+        ]
+        
+        replacements.forEach({
+            alteredSource = alteredSource.replacingOccurrences(of: $0.key, with: $0.value, options: .literal)
+        })
+        return WKUserScript(source: alteredSource, injectionTime: .atDocumentStart, forMainFrameOnly: false)
+    }()
+    
+    private var handler: (PlaylistInfo?) -> Void = { _ in }
+    private var handlerDidExecute = false
+    
+    init(handler: @escaping (PlaylistInfo?) -> Void) {
+        super.init(frame: .zero)
+
+        guard let webView = tab.webView else {
+            handlerDidExecute = true
+            handler(nil)
+            return
+        }
+        
+        self.handler = { [weak self] in
+            guard let self = self else { return }
+            self.handlerDidExecute = true
+            handler($0)
+        }
```

Comment:
This init is confusing:

1. Why are we tracking `handlerDidExecute`? Add doc if its necessary, but I only see it referenced in one spot so it seems weird.
2. Why does `handler` have a default empty block at its definition? 
3. Why not make this an optional init and return nil instead of returning early but retaining the init? Perhaps doing this will let you remove the initial value of `handler`

Reply:
I fixed it. I was already using a dispatch work item to cancel anyway so it's not needed to track it. I still do have to call the handler if I can't initialize properly so that it causes the media item to show as expired.